tests/lib/eslint/flat-eslint.js

```diff
@@ -5033,17 +5027,17 @@ describe("FlatESLint", () => {
             });
         });
 
-        // dependent on https://github.com/mrmlnc/fast-glob/issues/86
-        xdescribe("if { ignores: 'foo/*.js', ... } is present by '--config node_modules/myconf/eslint.config.js',", () => {
+        describe("if { ignores: 'foo/*.js', ... } is present by '--config node_modules/myconf/eslint.config.js',", () => {
             const { prepare, cleanup, getPath } = createCustomTeardown({
                 cwd: `${root}a3`,
                 files: {
-                    "node_modules/myconf/eslint.config.js": `module.exports = {
-                        ignores: ["**/eslint.config.js", "!node_modules/myconf", "foo/*.js"],
+                    "node_modules/myconf/eslint.config.js": `module.exports = [{
+                        ignores: ["!node_modules/myconf", "foo/*.js"],
```

Comment:
`!node_modules/myconf` also shouldn't have any effects after `**/node_modules/**`. because it unignores the directory but everything under it is still ignored by `**/node_modules/**`.

Reply:
This also seems counter to what existing tests are indicating. 🤔

Reply:
Here is the original test:
https://github.com/eslint/eslint/blob/main/tests/lib/eslint/eslint.js#L6426-L6459

It seems to behave in the same way as I have currently.

Reply:
> Here is the original test: https://github.com/eslint/eslint/blob/main/tests/lib/eslint/eslint.js#L6426-L6459

This is a bit different because in eslintrc the default pattern is [`/**/node_modules/*`](https://github.com/eslint/eslintrc/blob/5438c6ba1e01fdfb02a7f09fafb73e6dabe31c2d/lib/config-array/ignore-pattern.js#L113) (`*` instead of `**` at the end is the difference; leading slash doesn't matter), so in the original test the list of patterns is:

```
/**/node_modules/*
!/node_modules/myconf
```

`node_modules/` doesn't match any patterns; `node_modules/myconf/` matches `/**/node_modules/*` but also matches a subsequent negative pattern  `!/node_modules/myconf` so it's not ignored;  `node_modules/myconf/foo/` doesn't match any patterns; `node_modules/myconf/foo/test.js` doesn't match any patterns.

In the flat eslint test, the list of patterns is:

```
**/node_modules/**
!node_modules/myconf
```

`node_modules/` doesn't match any patterns; `node_modules/myconf/` matches `**/node_modules/*` but also matches a subsequent negative pattern `!node_modules/myconf` so it's not ignored; `node_modules/myconf/foo/` matches `**/node_modules/**` and doesn't match any negative patterns so that should be the end - directory `node_modules/myconf/foo/` is ignored and thus everything under it is ignored as well, including `node_modules/myconf/foo/test.js`.